### PR TITLE
St/news box

### PR DESCRIPTION
**Pull request by stevethomas19**
This is the new area where people can edit the news box found [here](http://screencast.com/t/ikgdbnhl0u).

Things I still need to do:
1. Add tests.
2. Make sure that the query in hlr is pulling from the hlrdesk db.

### CODE REVIEW

File: core/db/migration/migrate-db.sh [hunk 1]
```diff
@@ -16,8 +16,14 @@ echo "Implementing Migrations"
 
 psql -c "select addcol('public','users', 'last_login', 'timestamp', 'current_timestamp');"
 psql -c "select addcol('public','media', 'fine_amount', 'real', '0.50');"
+psql -c "CREATE TABLE newsbox (
```
Comment (trevren11) [inline core/db/migration/migrate-db.sh]: @stevethomas19  Will this rewrite or write over this table every time we run this?

**Comment by stevethomas19** [comment]
@trevren11 Good question. To be fully honest I don't understand the purpose of this file. I don't understand the format of the select addcol or even if we are supposed to add the new table here.

File: core/db/migration/migrate-db.sh (hunk 1 above)
Comment (webnard) [inline core/db/migration/migrate-db.sh]: @trevren11 This won't harm the table if it exists, but it will throw an error. Maybe put an `IF NOT EXISTS` clause so it keeps quiet?

**Comment by webnard** [comment]
Looks good. I haven't played with it—just looked at the code—so you might want to walk someone through its use when you get the chance.

In the future you'll probably want to use the template tag to make the JavaScript work less onerous: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template. Since we're only targeting modern browsers with the possible exception of the calendar, feel free to use any cutting edge features.

**Comment by stevethomas19** [comment]
Ok we have an RSS feed that works! When we are ready to implement these changes live (or when there is news in the production db) this is the code to put in news.php at hlr.byu.edu.

```php
<?php

$xml = file_get_contents('http://hlrdesk.byu.edu/rss');

$x = new SimpleXMLElement($xml);


echo "<div id=\"news\">
        <h2>HLR News</h2>";
foreach($x->channel->item as $entry) {
        echo "<div>
        <h3>$entry->title</h3>
        <a href='$entry->link' title=\"Image retrieved from $entry->link\"><img src='$entry->link' alt=\"image representing this news\" /></a>
        <p>$entry->description</p>
        </div>";
}
echo "</div>";
?>
```

For testing purposes do the following:
1. Pull the code and npm install
2. Add news article(s) to the db
3. ssh to hlr.byu.edu and edit news.php to look like the following code
      a) instead of `'http://hlrdesk.byu.edu/rss'` use your hostname i.e `'http://Stephens-Air.app.byu.edu:8080/rss'` 
4. navigate to hlr.byu.edu - you should see your news in the News box on the live site.

File: core/app.js [hunk 2]
```diff
@@ -332,6 +332,32 @@ app.use(_.get("/employees",function *(){
   }
 }));
 
+app.use(_.get("/rss",function *(){
+  var rss = require('rss');
+  var newsbox = yield require('./app_modules/newsbox').list;
+  var feed = new rss({
+    title: 'NewsBox',
+    feed_url: 'http://' + ENV.HLRDESK_HOST + ENV.PORT + '/rss',
```
Comment (webnard) [inline core/app.js]: This URL will be incorrect on the production site, since the port will be an internal-only port, and it's running on HTTPS.

Use app_modules/utils.js for an example of generating an appropriate URL. For example:

`utils.gen_url(ENV.HLRDESK_HOST, ENV.PORT, '/rss', !ENV.HLRDESK_DEV)`

File: public/js/newsbox.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  addNews.onclick = function() {
+    document.getElementById('addNews').disabled = "disabled";
+    newsEditTable.insertRow(-1);
+    var last_row = newsEditTable.rows[ newsEditTable.rows.length - 1];
+    for (var i = 0; i < 4; i++){
+      var cell = last_row.insertCell(-1);
+      switch(i) {
+        case 0:
+          cell.innerHTML = '<input id="new-img-link" type=text required value="">'
```
Comment (webnard) [inline public/js/newsbox.js]: Here you might consider using a `<template>` tag to improve readability.

**Comment by webnard** [comment]
The main thing I see is just a potentially inappropriate URL being used for the RSS feed. It likely won't affect anything, but it's worth looking into since it's an easy fix. Once that's done, I'd say merge.

**Comment by webnard** [comment]
Barring the merge conflict, everything LGTM.

File: core/db/migration/migrate-db.sh [hunk 4]
```diff
@@ -18,12 +18,6 @@ psql -c "select addcol('public','users', 'last_login', 'timestamp', 'current_tim
 psql -c "select addcol('public','media', 'fine_amount', 'real', '0.50');"
 psql -c "select addcol('public','media', 'code', 'varchar(2)', null);"
 psql -c "select addcol('public','inventory', 'icn', 'varchar(16)', null);"
-psql -c "CREATE TABLE newsbox (
```
Comment (webnard) [inline core/db/migration/migrate-db.sh]: Why was this removed?
